nuxeo-features/nuxeo-platform-mail/nuxeo-platform-mail-core/src/main/java/org/nuxeo/ecm/platform/mail/listener/action/CreateDocumentsFromAutomationChainAction.java

```diff
@@ -86,27 +85,30 @@ public boolean execute(ExecutionContext context) {
 
         AutomationService as = Framework.getService(AutomationService.class);
 
-        OperationContext automationCtx = new OperationContext(session);
-        automationCtx.putAll(context);
-
-        ExecutionContext initialContext = context.getInitialContext();
-        String parentPath = (String) initialContext.get(PARENT_PATH_KEY);
-        DocumentModel mailFolder = session.getDocument(new PathRef(parentPath));
-        automationCtx.put("mailFolder", mailFolder);
-        automationCtx.put("executionContext", initialContext);
-        String subject = (String) context.get(SUBJECT_KEY);
-        automationCtx.put("mailDocumentName", generateMailName(subject));
-
-        @SuppressWarnings("unchecked")
-        List<FileBlob> attachments = (List<FileBlob>) context.get(ATTACHMENTS_KEY);
-        if (attachments == null) {
-            automationCtx.put(ATTACHMENTS_KEY, Collections.EMPTY_LIST);
-        }
-
-        try {
-            as.run(automationCtx, getChainName());
-        } catch (OperationException e) {
-            throw new NuxeoException(e);
+        try (OperationContext automationCtx = new OperationContext(session)) {
+            automationCtx.putAll(context);
+
+            ExecutionContext initialContext = context.getInitialContext();
+            String parentPath = (String) initialContext.get(PARENT_PATH_KEY);
+            DocumentModel mailFolder = session.getDocument(new PathRef(parentPath));
+            automationCtx.put("mailFolder", mailFolder);
+            automationCtx.put("executionContext", initialContext);
+            String subject = (String) context.get(SUBJECT_KEY);
+            automationCtx.put("mailDocumentName", generateMailName(subject));
+
+            @SuppressWarnings("unchecked")
+            List<FileBlob> attachments = (List<FileBlob>) context.get(ATTACHMENTS_KEY);
+            if (attachments == null) {
+                automationCtx.put(ATTACHMENTS_KEY, Collections.EMPTY_LIST);
+            }
+
+            try {
+                as.run(automationCtx, getChainName());
+            } catch (OperationException e) {
+                throw new NuxeoException(e);
+            }
+        } catch (Exception cause) {
```

Comment:
`cause` —> `e` and call `ExceptionUtils.checkInterrupt(e)` immediately after the `catch (Exception ...)`